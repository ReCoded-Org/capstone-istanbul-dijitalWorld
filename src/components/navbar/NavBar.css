.navbar {
  padding: 0;
  height: 5em;
  width: 100%;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.08);
}

/* bootstrap class  */
.navbarMainContent.navbar-toggler.collapsed {
  position: absolute;
  margin-left: 5%;
}

/* bootstrap class */
.navbar-light .navbar-nav .nav-link {
  color: #6c567b !important;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 16px;
}

/* bootstrap class */
.navbar-collapse {
  justify-content: flex-end;
}

/* bootstrap class */
.navbar-light .navbar-nav .nav-link:hover {
  color: #f8b195 !important;
  justify-content: flex-end;
}

.navbarBrand {
  margin-left: 3em;
}

.brandLogo {
  height: 70px;
  width: 70px;
}

.buttonGroup {
  margin-left: 11em;
}

.signupFilledButton,
.loginOutlinedButton {
  box-shadow: 0px 10px 20px rgba(49, 49, 49, 0.116);
  font-size: 14px;
  display: inline-flex;
  justify-content: center;
  margin: 0.3em;
  border-radius: 40px !important;
}

.signupFilledButton {
  background-color: #6c567b !important;
  color: #fff !important;
  outline: none;
  border: none !important;
}

.loginOutlinedButton {
  color: #f8b195 !important;
  border-color: #f8b195 !important;
}

.signupFilledButton:hover {
  background-color: #fff !important;
  color: #6c567b !important;
  border: #6c567b 1px solid !important;
}

.signupFilledButton:active {
  color: #fff;
  outline: #6c567b 1px solid;
  color: #6c567b;
  box-shadow: none;
}

.signupFilledButton:focus {
  background-color: #6c567b;
  color: #fff;
  box-shadow: none;
}

.loginOutlinedButton:hover {
  color: #fff !important;
  background-color: #f8b195;
}

/* bootstrap class */
.dropdown-toggle {
  background-color: transparent;
  width: auto;
  font-size: 12px;
  border: none;
  color: #6c567b;
}

/* bootstrap class */
.dropdown-toggle:active {
  box-shadow: none !important;
  border: none !important;
  outline: none !important;
}

/* bootstrap class */
.dropdown-toggle::after {
  color: #f8b195;
  margin-right: 3em;
}

/* bootstrap class */
.dropdown-menu {
  font-size: 12px;
}

@media only screen and (min-device-width: 998px) and (max-device-width: 1024px) {
  .navLinksContainer {
    margin-left: 15em;
  }

  /* bootstrap class */
  .dropdown-toggle::after {
    margin-right: 0px;
  }

  .navbarBrand {
    margin-left: 30px;
  }

  .buttonGroup {
    margin-left: 4em;
  }

  .signupFilledButton,
  .loginOutlinedButton {
    box-shadow: 0px 10px 20px rgba(49, 49, 49, 0.2);
    font-size: 10px;
    width: 80px;
    height: 30px;
    margin: 0.3em;
    border-radius: 40px;
  }
}

@media only screen and (min-device-width: 850px) and (max-device-width: 998px) {
  /* bootstrap class */
  .dropdown-toggle::after {
    margin-right: 0px;
  }

  .navbarBrand {
    margin-left: 50px;
  }

  .buttonGroup {
    margin-left: 5em;
  }

  .signupFilledButton,
  .loginOutlinedButton {
    box-shadow: 0px 10px 20px rgba(49, 49, 49, 0.2);
    font-size: 10px;
    width: 70px;
    height: 30px;
    margin: 0.3em;
    border-radius: 40px;
  }
}

@media only screen and (min-device-width: 576px) and (max-device-width: 850px) {
  /* bootstrap class */
  .dropdown-toggle::after {
    margin-right: 0px;
  }

  .navbarBrand {
    margin: auto;
  }

  .navLinkText {
    font-size: 12px !important;
  }

  .buttonGroup {
    margin-left: 1em;
  }

  /* bootstrap class */
  .navbar-collapse {
    justify-content: space-around !important;
  }

  .signupFilledButton,
  .loginOutlinedButton {
    box-shadow: 0px 10px 20px rgba(49, 49, 49, 0.2);
    font-size: 9px;
    width: 60px;
    height: 20px;
    margin: 0.2em;
    border-radius: 40px;
  }
}

.userPhoto:hover {
  cursor: pointer;
}
