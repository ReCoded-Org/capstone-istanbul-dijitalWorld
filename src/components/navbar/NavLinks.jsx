import React from 'react';
import { Nav } from 'react-bootstrap/';
import { useTranslation } from 'react-i18next';
import './NavBar.css';

export default function NavLinks() {
  const { t } = useTranslation();

  return (
    <>
      <Nav.Link className="navLinkText" href="#home">
        {t('home.navBar.navLinks.home')}
      </Nav.Link>
      <Nav.Link className="navLinkText" href="#home">
        {t('home.navBar.navLinks.blog')}
      </Nav.Link>
      <Nav.Link className="navLinkText" href="#home">
        {t('home.navBar.navLinks.inspire')}
      </Nav.Link>
      <Nav.Link className="navLinkText" href="#home">
        {t('home.navBar.navLinks.resource')}
      </Nav.Link>
      <Nav.Link className="navLinkText" href="/about">
        {t('home.navBar.navLinks.about')}
      </Nav.Link>
      <Nav.Link className="navLinkText" href="#home">
        {t('home.navBar.navLinks.contact')}
      </Nav.Link>
    </>
  );
}
