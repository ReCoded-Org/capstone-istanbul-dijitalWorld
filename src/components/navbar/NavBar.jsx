import React, { useState, useEffect } from 'react';
import { Navbar, Nav, Dropdown, DropdownButton, Button, Image } from 'react-bootstrap/';
import { useDispatch, useSelector } from 'react-redux';
import { userLoggedInAction } from '../../redux/action';
import { auth } from '../../firebase';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';
import NavLinks from './NavLinks';
import { Route } from 'react-router-dom';
import './NavBar.css';

const ANONYMOUS_PHOTO_URL =
  'https://st3.depositphotos.com/4111759/13425/v/450/depositphotos_134255710-stock-illustration-avatar-vector-male-profile-gray.jpg';

export default function NavBar({ routes }) {
  const dispatch = useDispatch();
  // Incase user is logged out this variable will be null which will case the user picture not to render
  const userData = useSelector((state) => state.currentUserDataReducer);
  const [isLoggedOut, setIsLoggedOut] = useState(false);

  const loginSignupButton = (
    <>
      <Route
        render={({ history }) => (
          <Button className="signupFilledButton" onClick={() => history.push('/signup')}>
            {t('home.navBar.buttons.signup')}
          </Button>
        )}
      ></Route>
      <Route
        render={({ history }) => (
          <Button
            className="loginOutlinedButton"
            variant="outline"
            onClick={() => history.push('/login')}
          >
            {t('home.navBar.buttons.login')}
          </Button>
        )}
      />
    </>
  );

  const CustomToggle = React.forwardRef(({ children, onClick }, ref) => (
    <>
      <Image
        className="userPhoto"
        onClick={(e) => {
          e.preventDefault();
          onClick(e);
        }}
        src={userData.photoURL || ANONYMOUS_PHOTO_URL}
        roundedCircle
        style={{ width: '50px' }}
      />
    </>
  ));

  useEffect(() => {
    auth.onAuthStateChanged((user) => {
      if (user) {
        setIsLoggedOut(false);
        // Providerdata's value contains an array that has the user's info in the first index as an object
        dispatch(userLoggedInAction(user.providerData[0]));
      } else {
        dispatch(userLoggedInAction(null));
        setIsLoggedOut(true);
      }
    });
  });

  const { t } = useTranslation();

  return (
    <Navbar className="navbar" data-testid="navbar" expand="sm">
      <Navbar.Brand className="navbarBrand" href="#home">
        <img
          data-testid="brand-logo"
          src={require('../../images/www-logo.png')}
          className="brandLogo d-inline-block align-top"
          alt="World with Women Logo"
        />
      </Navbar.Brand>
      <Navbar.Toggle className="navbarMainContent" aria-controls="basic-navbar-nav" />
      <Navbar.Collapse data-testid="navbar-collapse" id="basic-navbar-nav">
        <Nav className="navLinksContainer">
          <NavLinks routes={routes} />
        </Nav>
        <div className="buttonGroup">
          {userData && (
            <Dropdown alignRight className="mr-3">
              <Dropdown.Toggle as={CustomToggle} id="dropdown-split-basic"></Dropdown.Toggle>
              <Dropdown.Menu>
                <Route
                  render={({ history }) => (
                    <Dropdown.Item onClick={() => history.push('/profile')} eventKey="1">
                      Profile
                    </Dropdown.Item>
                  )}
                />
                <Route
                  render={({ history }) => (
                    <Dropdown.Item
                      onClick={() => {
                        auth.signOut();
                        history.push('/');
                      }}
                      eventKey="1"
                    >
                      Sign out
                    </Dropdown.Item>
                  )}
                />
              </Dropdown.Menu>
            </Dropdown>
          )}
          {isLoggedOut && loginSignupButton}
        </div>
      </Navbar.Collapse>
      <DropdownButton
        alignRight
        variant="none"
        id="dropdown-item-button"
        className="dropdownItemButton"
        title="EN"
      >
        <Dropdown.Item as="button">EN</Dropdown.Item>
        <Dropdown.Item as="button">TR</Dropdown.Item>
        <Dropdown.Item as="button">AR</Dropdown.Item>
      </DropdownButton>
    </Navbar>
  );
}

NavBar.propTypes = {
  routes: PropTypes.arrayOf(PropTypes.object),
};

NavBar.defaultProps = {
  routes: [
    {
      path: '/',
      name: 'home',
    },
  ],
};
