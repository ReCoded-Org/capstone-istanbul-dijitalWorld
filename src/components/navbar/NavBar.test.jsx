import React from "react";
import ReactDOM from "react-dom";
import "@testing-library/jest-dom";
import NavBar from "./NavBar";
import NavLinks from "./NavLinks";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<NavBar />, div);
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<NavLinks />, div);
});
