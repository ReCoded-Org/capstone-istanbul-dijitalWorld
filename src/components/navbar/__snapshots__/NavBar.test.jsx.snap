// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className="navbar navbar navbar-expand-sm navbar-light"
  data-testid="navbar"
>
  <a
    className="navbarBrand navbar-brand"
    href="#home"
  >
    <img
      alt="World with Women Logo"
      className="brandLogo d-inline-block align-top"
      data-testid="brand-logo"
      src="www-logo.png"
    />
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbarMainContent navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    data-testid="navbar-collapse"
    id="basic-navbar-nav"
  >
    <div
      className="navLinksContainer navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="navLinkText nav-link"
        data-rb-event-key="/"
        href="/"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        home.navBar.navLinks.home
      </a>
      <a
        className="navLinkText nav-link"
        data-rb-event-key="/blog"
        href="/blog"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        home.navBar.navLinks.blog
      </a>
      <a
        className="navLinkText nav-link"
        data-rb-event-key="/inspire"
        href="/inspire"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        home.navBar.navLinks.inspire
      </a>
      <a
        className="navLinkText nav-link"
        data-rb-event-key="/resources"
        href="/resources"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        home.navBar.navLinks.resource
      </a>
      <a
        className="navLinkText nav-link"
        data-rb-event-key="/about"
        href="/about"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        home.navBar.navLinks.about
      </a>
      <a
        className="navLinkText nav-link"
        data-rb-event-key="/contact"
        href="/contact"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        home.navBar.navLinks.contact
      </a>
    </div>
    <div
      className="buttonGroup"
    >
      <button
        className="signupFilledButton btn btn-primary"
        disabled={false}
        type="button"
      >
        home.navBar.buttons.signup
      </button>
      <button
        className="loginOutlinedButton btn btn-outline"
        disabled={false}
        type="button"
      >
        home.navBar.buttons.login
      </button>
    </div>
  </div>
  <div
    className="dropdownItemButton dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-none"
      disabled={false}
      id="dropdown-item-button"
      onClick={[Function]}
      type="button"
    >
      EN
    </button>
  </div>
</nav>
`;
